presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayDistinctFunction.java

```diff
@@ -58,62 +52,133 @@ public Block distinct(@TypeParameter("E") Type type, @SqlType("array(E)") Block
             }
         }
 
-        TypedSet typedSet = new TypedSet(type, array.getPositionCount(), "array_distinct");
-        int distinctCount = 0;
+        TypedSet typedSet = new TypedSet(type, arrayLength, "array_distinct");
+        BlockBuilder distinctElementBlockBuilder;
+        int position = 0;
```

Comment:
nit: what about `firstDuplicatePosition`?

Reply:
ping @kaikalur

Reply:
It's actually current position, not necessarily distinct.